pwiz_tools/Skyline/Controls/Graphs/GraphFullScan.cs

```diff
@@ -478,30 +610,49 @@ private void CreateSingleScan()
                 assignedPointList.Add(mz, intensity);
             }
 
-            // Create a graph item for each point list with its own color.
-            for (int i = 0; i < pointLists.Length; i++)
-            {
-                var transition = _msDataFileScanHelper.ScanProvider.Transitions[i];
-                if (transition.Source != _msDataFileScanHelper.Source)
-                    continue;
-                var item = new SpectrumItem(pointLists[i], transition.Color, _msDataFileScanHelper.ScanProvider.Transitions[i].Name, 2);
-                var curveItem = _graphHelper.GraphControl.AddGraphItem(GraphPane, item, false);
-                curveItem.Label.IsVisible = false;
-            }
+            var selection = GraphSpectrum.SpectrumNodeSelection.GetCurrent(Program.MainWindow);
+            //find out if the current selection belongs to the same precursor as the loaded MS spectrum
+            var dataPrecursor = _msDataFileScanHelper.ScanProvider.Transitions.FirstOrDefault(t => (t.Id as Transition)?.IonType == IonType.precursor);
+            var selectionMatch = ReferenceEquals(selection.Precursor.Id, (dataPrecursor?.Id as Transition)?.Group);

```

Comment:
I get a NullReferenceException when I click on peptides in the Targets tree.
I think this needs to be:
selection.Precursor?.Id

Reply:
Good catch. Fixed, thank you!